### DISCUSSION TITLE
`use "begin" instead` error when trying to execute async method in Python

### DISCUSSION THREAD

**Discussion by zhzhang** [aws/jsii#3460] [Q&A]
### Affected Languages

- [ ] `TypeScript` or `Javascript`
- [X ] `Python`
- [ ] `Java`
- [ ] .NET (`C#`, `F#`, ...)
- [ ] `Go`

### General Information
* **JSII Version: 1.34.0** 
* **Platform: OSX**     

### The Question
I have an async class method defined like
```
export class ProjectResources {
    ...
  public static async createProjectResources(
    props: ProjectResourcesProps
  ): Promise<ProjectResources> {
    // Some more async calls here.
  }
}
```
That is producing `jsii.errors.JSIIError: createProjectResources is an async method, use "begin" instead` when it is executed in python. 
Is there any way I should be defining or using async methods in python differently?

The TS code compiles to Python and packages without any errors.

**Comment by zhzhang** [comment]
Looking at the generated files, it looks like the `static` keyword is causing the invocation to use `sinvoke` instead of `ainvoke`.

**Comment by automartin5000** [comment]
I'm running into this same issue. Did you ever find a solution?

**Comment by XiangGu97** [comment]
Run into the same issue as well, I cannot find any instruction to tell me how to use this `begin`.
